cabal-install-solver/src/Distribution/Solver/Types/PkgConfigDb.hs

```diff
@@ -116,17 +109,17 @@ readPkgConfigDb verbosity progdb = handle ioErrorHandler $ do
           -- requested one, we fall back to querying one by one.
           do
             info verbosity ("call to pkg-config --modversion on all packages failed. Falling back to querying pkg-config individually on each package")
-            pkgConfigDbFromList . catMaybes <$> mapM (getIndividualVersion pkgConfig) pkgNames
+            Just . pkgConfigDbFromList . catMaybes <$> mapM (getIndividualVersion pkgConfig) pkgNames
   where
     -- For when pkg-config invocation fails (possibly because of a
     -- too long command line).
     noPkgConfig extra = do
         info verbosity ("Failed to query pkg-config, Cabal will continue"
                         ++ " without solving for pkg-config constraints: "
```

Comment:
This still needs to be updated. We do solve for the pkg-config constraints (we did even before this PR).